tests/core/editable/keystrokes/delbackspacequirks/collapsed.js

```diff
@@ -33,6 +33,62 @@
 
 		// --- MISC -----------------------------------------------------------
 
+		// (#3819)
+		'test backspace key use when carret between two visual spaces did not split content': function() {
+			bender.editorBot.create( {
+				name: 'editor' + new Date().getTime()
+			}, function( bot ) {
+				// Intentionally makes selection marker(`{}`) at the end.
+				// If we put it between two spaces at the beginning - content will be splited before key simulation.
+				bender.tools.selection.setWithHtml( bot.editor, 'Hello&nbsp; World{}' );
+
+				var editable = bot.editor.editable(),
+					pContent = editable.findOne( 'p' ),
+					range = bot.editor.createRange();
+
+				// Move selection between two visual spaces.
+				range.setStart( pContent.getFirst(), 6 );
+				range.select();
+
+				editable.fire( 'keydown', new CKEDITOR.dom.event( { keyCode: BACKSPACE } ) );
+
+				var foundIndex = CKEDITOR.tools.getIndex( pContent.getChildren().toArray(), function( node ) {
+					var firstCharCode = node.getText().charCodeAt( 0 );
+					return firstCharCode === 32 || firstCharCode === 160;
+				} );
+
+				assert.areEqual( -1, foundIndex, 'There should not be a node starting with whitespace after backspace key' );
+			} );
```

Comment:
Both unit tests are almost the same so we could extract them to the helper function.